### PR TITLE
Fix links in changelog entries

### PR DESCRIPTION
**Pull request by namurphy**
This PR makes a few minor changes to changelog entries like fixing broken reST links.  Should be pretty quick.

### CODE REVIEW

**Comment by namurphy** [comment]
I still need to double check the links on the page, and add links to a few pages in the docs that I didn't add in the original PRs because I had forgotten the reST syntax and didn't feel like looking up (sigh).

 - [ ] Add links to the testing guide & contributor guide for 1316
 - [ ] Link to documentation guide for 1373
 - [ ] Link to bibliography for 1374

**Comment by namurphy** [comment]
I used the "no changelog entry needed" because the Giles check wasn't passing, but there actually is a changelog entry for this PR.

File: changelog/1316.doc.rst [hunk 1]
```diff
@@ -1,2 +1,2 @@
 Added a discussion of test parametrization with argument unpacking to
-the testing guide in the contributor guide.
+the testing guide in the |contributor guide|.
```
Comment (rocco8773) [inline changelog/1316.doc.rst:2]: This is currently not render correctly on RTD.

image

File: changelog/1373.doc.rst [hunk 2]
```diff
@@ -1,2 +1,2 @@
 Described additional environments for building the documentation with make_
-in the documentation guide.
+in the |documentation guide|.
```
Comment (rocco8773) [inline changelog/1373.doc.rst:2]: Not rendering correctly on RTD.

image

File: changelog/1374.doc.rst [hunk 3]
```diff
@@ -1 +1 @@
-Moved references from individual docstrings to the bibliography.
+Moved references from individual docstrings to the |bibliography|.
```
Comment (rocco8773) [inline changelog/1374.doc.rst:1]: Also not not rendering correctly on RTD.

image

**Review by rocco8773 — APPROVED** [review]
LGTM!!

**Pull request merged**